extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchHolderObjectStrategy.java

```diff
@@ -67,4 +74,32 @@ public HllSketchHolder fromByteBufferSafe(ByteBuffer buffer, int numBytes)
         )
     );
   }
+
+  private boolean isSafeToConvertToNullSketch(ByteBuffer buf)
+  {
+    // TODO: Might need a sanity check here, to ensure that position and offset makes sense.
+
+    // Get org.apache.datasketches.hll.CurMode. This indicates the type of data structure.
+    final int position = buf.position();
+    final int preInts = buf.get(position) & 0X3F; // get(PREAMBLE_INTS_BYTE) & 0X3F
+    final int curMode = buf.get(position + 7) & 3;    // get(MODE_BYTE) & CUR_MODE_MASK
+
+    switch (curMode) {
+      case 0:          // LIST
+        Preconditions.checkArgument(preInts == 2);  // preInts == LIST_PREINTS, Sanity
```

Comment:
`Preconditions.checkArgument` doesn't generate a nice exception for an end user and, this usage of `Preconditions` from Guava is one of the sources of code-level incompatibility between versions of Guava.  In general, it's best to avoid dependence on Guava wherever possible.  For checking an argument like this, it might make sense to add a static method on `DruidException` or have some other utility type place that will do the check and throw a `DruidException` instead of the generic `IllegalArgumentException`.

That said, throwing an exception in this case seems overly zealous.  It seems better to just return false if our assumption is broken here and let the underlying HLL library do what it will with it.

Reply:
Changed the code here to return false instead. I think the point about using DruidException instead of Guava makes sense.